src/main/java/games/strategy/engine/data/GameDataMemento.java

```diff
@@ -101,11 +101,16 @@ private GameDataMemento() {}
 
     @Override
     public void exportProperties(final GameData gameData, final Map<String, Object> propertiesByName) {
-      propertiesByName.put(PropertyNames.DICE_SIDES, gameData.getDiceSides());
```

Comment:
I'm assuming you meant to tag **new** line 104 below (the `acquireReadLock()` call); please clarify if that's not the case.

The [standard idiom](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/Lock.html) for `Lock`-based structures is to acquire, do work in a `try` block, and release in the corresponding `finally`.  A search of our codebase shows that only 18 of 117 calls to `acquireReadLock()` and 2 of 9 calls to `acquireWriteLock()` do not follow this idiom.  That is, 84% of the existing code does follow this idiom.

I would argue that the `acquireXXXLock()` methods should never throw a recoverable exception.  Either they acquire the lock and return or do not acquire the lock and throw (and there's the third possibility that they may block forever if there's a deadlock).  I looked at the code in `LockUtil#acquireLock()` (to which both of the `GameData` methods delegate), and, in fact, that appears to be the case.  `Lock#lock()` is the very last statement in the method, so there's no possibility of an exception being thrown after the lock is acquired.